nncf/common/sparsity/controller.py

```diff
@@ -17,14 +19,21 @@ class SparsityController(CompressionAlgorithmController):
     This is the class from which all sparsity controllers inherit.
     """
 
-    def set_sparsity_level(self, sparsity_level: float):
+    def set_sparsity_level(self, sparsity_level: float) -> None:
         """
         Sets the sparsity level that should be applied to the model's weights.
 
         :param sparsity_level: Sparsity level that should be applied to the model's weights.
         """
 
-    def freeze(self):
+    def freeze(self) -> None:
         """
         Freezes all sparsity masks. Sparsity masks will not be trained after calling this method.
         """
+
+    @property
+    @abstractmethod
+    def current_sparsity_level(self) -> float:
+        """
+        Returns the current sparsity level of the underlying model.
+        """
```

Comment:
@alexsu52 this exposes a new property on the derived `nncf.torch.sparsity.base_algo.BaseSparsityAlgoController` which is marked as API, hence the API marker was (correctly) assigned to this PR. This seems to be a good property for the sparsity controllers, and helps resolve one of the original mypy issues with our code. Do you have any comments or objections for that?